### PR TITLE
feat: add HLS Streaming option 

### PR DESCRIPTION
**Pull request by a-vishnoi**

### CODE REVIEW

File: src/views/HLSView.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const defaultClasses = {
+  HLSVideo: "w-full contain",
+};
+
+export const HLSView = ({ isChatOpen, toggleChat, isParticipantListOpen }) => {
+  const videoRef = useRef(null);
+  const peers = useHMSStore(selectPeers);
+  const hlsState = useHMSStore(selectHLSState);
+  useEffect(() => {
+    if (Hls.isSupported()) {
```
Comment (raviteja83) [inline src/views/HLSView.jsx]: add null check for videoref.current. Make this if Hls.isSupported && hlsState.url

File: src/views/HLSView.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { getBlurClass } from "../common/utils";
+import { ChatView } from "./components/chatView";
+
+const defaultClasses = {
+  HLSVideo: "w-full contain",
+};
+
+export const HLSView = ({ isChatOpen, toggleChat, isParticipantListOpen }) => {
+  const videoRef = useRef(null);
+  const peers = useHMSStore(selectPeers);
```
Comment (triptu) [inline src/views/HLSView.jsx]: selectPeerCount

File: src/views/HLSView.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+} from "@100mslive/hms-video-react";
+import Hls from "hls.js";
+import { getBlurClass } from "../common/utils";
+import { ChatView } from "./components/chatView";
+
+const defaultClasses = {
+  HLSVideo: "w-full contain",
+};
+
+export const HLSView = ({ isChatOpen, toggleChat, isParticipantListOpen }) => {
```
Comment (triptu) [inline src/views/HLSView.jsx:17]: Let's show a message in case there is no hls url.

File: src/views/components/MoreSettings.jsx [hunk 4]
```diff
@@ -66,9 +69,15 @@ export const MoreSettings = () => {
...
   const [rtmpURL, setRtmpURL] = useState("");
+  const [isHlsOn, setIsHlsOn] = useState(false);
   const recording = useHMSStore(selectRecordingState);
   const rtmp = useHMSStore(selectRTMPState);
+  const hls = useHMSStore(selectHLSState);
+  const [prevRole, setPrevRole] = useState(localPeer.roleName);
   const [isRecordingOn, setIsRecordingOn] = useState(false);
+  const HLSRole =
+    roomMetadata[DEFAULT_HLS_ROLE_KEY] ||
+    process.env.REACT_APP_DEFAULT_HLS_ROLE;
```
Comment (triptu) [inline src/views/components/MoreSettings.jsx]: The redundant env variable can be removed. Room metadata will already have the role

**Comment by triptu** [comment]
Let's call this app details instead of room metadata, it aligns better with the dashboard api call and room metadata as a concept may actually come in future at which point it can cause confusion.

File: src/views/components/RecordingAndRTMPForm.jsx [hunk 5]
```diff
@@ -54,10 +57,39 @@ export const RecordingAndRTMPForm = ({
             />
           </div>
         </div>
+
+        <div className={defaultClasses.formInner}>
```
Comment (triptu) [inline src/views/components/RecordingAndRTMPForm.jsx]: The form component is getting quite big. Let's break it down by creating sub components in this file

File: src/views/components/MoreSettings.jsx [hunk 6]
```diff
@@ -149,6 +165,18 @@ export const MoreSettings = () => {
     }
   };
 
+  const switchView = async role => {
```
Comment (triptu) [inline src/views/components/MoreSettings.jsx]: All the state and logic related to rtmp, recording, hls can be refactored down

File: src/views/mainView.jsx [hunk 7]
```diff
@@ -26,7 +28,11 @@ export const ConferenceMainView = ({
   const peerSharingPlaylist = useHMSStore(selectPeerSharingVideoPlaylist);
   const roomState = useHMSStore(selectRoomState);
   const hmsActions = useHMSActions();
-  const { audioPlaylist, videoPlaylist, uiViewMode } = useContext(AppContext);
+  const { audioPlaylist, videoPlaylist, uiViewMode, roomMetadata } =
+    useContext(AppContext);
+  let HLSViewerRole =
+    roomMetadata[DEFAULT_HLS_ROLE_KEY] ||
```
Comment (triptu) [inline src/views/mainView.jsx]: Instead of having this at every place let the context return hlsViewerRole which can be directly used

**Review by triptu — DISMISSED** [review]

**Review by triptu — APPROVED** [review]

**Pull request merged**